Automattic/jetpack
# Add Cloudflare Analytics snippet to site header via (WPCOM only) Jetpack plugin

Fixes 406-gh-Automattic/dotcom-manage

#### Changes proposed in this Pull Request:

* This PR supports the needs of the Cloudflare partnership project (project thread at pbAPfg-1he-p2), which includes functionality that will need to be supported by both simple sites and Atomic sites (but not Jetpack standalone users). This plugin inserts a Cloudflare analytics JS snippet using a token stored as a user option; it should not run by default in Jetpack, but be activated from WPCOM code (see D56767-code).

#### Jetpack product discussion


N/A

#### Does this pull request change what data or activity we track or use?


No.

#### Testing instructions:






* Check out D56767-code in your sandbox.
* Copy the file in this PR into your sandbox at `wp-content/mu-plugins/jetpack/modules/cloudflare-analytics/class-wp-cloudflare-analytics.php`
* Set a Cloudflare analytics code as a site option.
    * In sandbox command line, run the command `wp shell --url={sandboxdomain} --user={a8cusername}`
    * From the shell, run the command `update_option('cloudflare_analytics', array('code' => '716d4d275266427cb78934b7919e73e5'));` to set the Cloudflare Analytics tracking ID for your blog.
* Navigate to the blog and verify that Cloudflare snippet has been added to the header.
![image](https://user-images.githubusercontent.com/13437011/107438386-01d27180-6af6-11eb-832e-828203d6b0a3.png)


#### Proposed changelog entry for your changes:


* Add Cloudflare analytics functionality to plugin to be used by WPCOM.
